File: win-prog/ProgWinCS/Menus/FirstMainMenu/FirstMainMenu.cs
//--------------------------------------------
// FirstMainMenu.cs © 2001 by Charles Petzold
//--------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class FirstMainMenu: Form
{
     public static void Main()
     {
          Application.Run(new FirstMainMenu());
     }
     public FirstMainMenu()
     {
          Text = "First Main Menu";

               // Items on File submenu

          MenuItem miOpen = new MenuItem("&Open...",
                                   new EventHandler(MenuFileOpenOnClick),
                                   Shortcut.CtrlO);

          MenuItem miSave = new MenuItem("&Save",
                                   new EventHandler(MenuFileSaveOnClick),
                                   Shortcut.CtrlS);

          MenuItem miSaveAs = new MenuItem("Save &As...",
                                   new EventHandler(MenuFileSaveAsOnClick));

          MenuItem miDash = new MenuItem("-");

          MenuItem miExit = new MenuItem("E&xit",
                                   new EventHandler(MenuFileExitOnClick));
               // File item

          MenuItem miFile = new MenuItem("&File",
                                   new MenuItem[] {miOpen, miSave, miSaveAs,
                                                   miDash, miExit });
               // Items on Edit submenu

          MenuItem miCut = new MenuItem("Cu&t",
                                   new EventHandler(MenuEditCutOnClick),
                                   Shortcut.CtrlX);

          MenuItem miCopy = new MenuItem("&Copy",
                                   new EventHandler(MenuEditCopyOnClick),
                                   Shortcut.CtrlC);

          MenuItem miPaste = new MenuItem("&Paste",
                                   new EventHandler(MenuEditPasteOnClick),
                                   Shortcut.CtrlV);
               // Edit item

          MenuItem miEdit = new MenuItem("&Edit",
                                   new MenuItem[] {miCut, miCopy, miPaste});

               // Item on Help submenu

          MenuItem miAbout = new MenuItem("&About FirstMainMenu...",
                                   new EventHandler(MenuHelpAboutOnClick));
               // Help item

          MenuItem miHelp = new MenuItem("&Help", 
                                   new MenuItem[] {miAbout});
               // Main menu

          Menu = new MainMenu(new MenuItem[] {miFile, miEdit, miHelp});
     }
     void MenuFileOpenOnClick(object obj, EventArgs ea)
     {
          MessageBox.Show("File Open item clicked!", Text);
     }
     void MenuFileSaveOnClick(object obj, EventArgs ea)
     {
          MessageBox.Show("File Save item clicked!", Text);
     }
     void MenuFileSaveAsOnClick(object obj, EventArgs ea)
     {
          MessageBox.Show("File Save As item clicked!", Text);
     }
     void MenuFileExitOnClick(object obj, EventArgs ea)
     {
          Close();
     }
     void MenuEditCutOnClick(object obj, EventArgs ea)
     {
          MessageBox.Show("Edit Cut item clicked!", Text);
     }
     void MenuEditCopyOnClick(object obj, EventArgs ea)
     {
          MessageBox.Show("Edit Copy item clicked!", Text);
     }
     void MenuEditPasteOnClick(object obj, EventArgs ea)
     {
          MessageBox.Show("Edit Paste item clicked!", Text);
     }
     void MenuHelpAboutOnClick(object obj, EventArgs ea)
     {
          MessageBox.Show(Text + " © 2001 by Charles Petzold");
     }
}

File: win-prog/ProgWinVB/Menus/FirstMainMenu/FirstMainMenu.vb
'----------------------------------------------
' FirstMainMenu.vb (c) 2002 by Charles Petzold
'----------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class FirstMainMenu
    Inherits Form

    Shared Sub Main()
        Application.Run(new FirstMainMenu())
    End Sub

    Sub New()
        Text = "First Main Menu"

        ' Items on File submenu
        Dim miOpen As New MenuItem("&Open...", _
                            AddressOf MenuFileOpenOnClick, _
                            Shortcut.CtrlO)

        Dim miSave As New MenuItem("&Save", _
                            AddressOf MenuFileSaveOnClick, _
                            Shortcut.CtrlS)

        Dim miSaveAs As New MenuItem("Save &As...", _
                            AddressOf MenuFileSaveAsOnClick)

        Dim miDash As New MenuItem("-")
        Dim miExit As New MenuItem("E&xit", _
                            AddressOf MenuFileExitOnClick)

        ' File menu item
        Dim miFile As New MenuItem("&File", _
                            New MenuItem() {miOpen, miSave, miSaveAs, _
                                            miDash, miExit})
        ' Items on Edit submenu
        Dim miCut As New MenuItem("Cu&t", _
                            AddressOf MenuEditCutOnClick, _
                            Shortcut.CtrlX)

        Dim miCopy As New MenuItem("&Copy", _
                            AddressOf MenuEditCopyOnClick, _
                            Shortcut.CtrlC)

        Dim miPaste As New MenuItem("&Paste", _
                            AddressOf MenuEditPasteOnClick, _
                            Shortcut.CtrlV)

        ' Edit menu item
        Dim miEdit As New MenuItem("&Edit", _
                            New MenuItem() {miCut, miCopy, miPaste})

        ' Item on Help submenu
        Dim miAbout As New MenuItem("&About FirstMainMenu...", _
                            AddressOf MenuHelpAboutOnClick)

        ' Help menu item
        Dim miHelp As New MenuItem("&Help", New MenuItem() {miAbout})

        ' Main menu
        Menu = new MainMenu(new MenuItem() {miFile, miEdit, miHelp})
    End Sub

    Private Sub MenuFileOpenOnClick(ByVal obj As Object, _
                                    ByVal ea As EventArgs)
        MessageBox.Show("File Open item clicked!", Text)
    End Sub

    Private Sub MenuFileSaveOnClick(ByVal obj As Object, _
                                    ByVal ea As EventArgs)
        MessageBox.Show("File Save item clicked!", Text)
    End Sub

    Private Sub MenuFileSaveAsOnClick(ByVal obj As Object, _
                                      ByVal ea As EventArgs)
        MessageBox.Show("File Save As item clicked!", Text)
    End Sub

    Private Sub MenuFileExitOnClick(ByVal obj As Object, _
                                    ByVal ea As EventArgs)
        Close()
    End Sub

    Private Sub MenuEditCutOnClick(ByVal obj As Object, _
                                   ByVal ea As EventArgs)
        MessageBox.Show("Edit Cut item clicked!", Text)
    End Sub

    Private Sub MenuEditCopyOnClick(ByVal obj As Object, _
                                    ByVal ea As EventArgs)
        MessageBox.Show("Edit Copy item clicked!", Text)
    End Sub

    Private Sub MenuEditPasteOnClick(ByVal obj As Object, _
                                     ByVal ea As EventArgs)
        MessageBox.Show("Edit Paste item clicked! ", Text)
    End Sub

    Private Sub MenuHelpAboutOnClick(ByVal obj As Object, _
                                     ByVal ea As EventArgs)
        MessageBox.Show(Text & " " & Chr(169) & " 2002 by Charles Petzold")
    End Sub
End Class
